src/aspire/volume/volume_synthesis.py

```diff
@@ -0,0 +1,303 @@
+import abc
+
+import numpy as np
+from numpy.linalg import qr
+
+from aspire.utils import Rotation, bump_3d, grid_3d
+from aspire.utils.random import Random, randn
+from aspire.volume import Volume
+
+
+class SyntheticVolumeBase(abc.ABC):
+    def __init__(self, L, C, symmetry, seed=None, dtype=np.float64):
+        self.L = L
+        self.C = C
+        self.symmetry = symmetry
+        self.seed = seed
+        self.dtype = dtype
+
+    @abc.abstractmethod
+    def generate(self):
+        """
+        Called to generate and return the synthetic volumes.
+
+        Each concrete subclass should implement this.
+        """
+
+    @abc.abstractmethod
+    def _check_symmetry(self):
+        """
+        Called to check that volumes are instantiated with compatible symmetry type.
+
+        Each concrete subclass should implement this.
+        """
+
+    def __repr__(self):
+        return f"{self.__dict__}"
```

Comment:
This is kind of cheezy.  It is good for prototyping, but we can probably dress it up and make something generalized and meaningful.. Also check that it doesn't crash when we call it.